File: src/main/java/hilfling/backend/hilfling/controllers/PurchaseOrderController.java
package hilfling.backend.hilfling.controllers;

import hilfling.backend.hilfling.model.PurchaseOrder;
import hilfling.backend.hilfling.service.PurchaseOrderService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/v1/purchase_orders")
public class PurchaseOrderController extends GenericBaseControllerImplementation<PurchaseOrder> {
    @Autowired
    private PurchaseOrderService service;
    public PurchaseOrderService getService() {
        return service;
    }
}

File: src/main/kotlin/hilfling/backend/hilfling/controller/PurchaseOrderController.kt
package hilfling.backend.hilfling.controller

import hilfling.backend.hilfling.model.Album
import hilfling.backend.hilfling.model.PurchaseOrder
import hilfling.backend.hilfling.repository.AlbumRepository
import hilfling.backend.hilfling.repository.PurchaseOrderRepository
import me.liuwj.ktorm.database.Database
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/purchase_orders")
class PurchaseOrderController {
    val repository = PurchaseOrderRepository()

    @GetMapping("/{id}")
    fun getById(@PathVariable("id") id: Int) : PurchaseOrder? {
        return repository.findById(id)
    }

    @GetMapping
    fun getAll() : List<PurchaseOrder> {
        return repository.findAll()
    }

    @PostMapping
    fun create(
            @RequestParam("purchaseOrder") purchaseOrder: PurchaseOrder
    ): PurchaseOrder {
        return repository.create(purchaseOrder)
    }
}